File: 0242-valid-anagram/0242-valid-anagram.java
// class Solution {
//     public boolean isAnagram(String s, String t) {
//         HashMap<Character,Integer> map = new HashMap<>();
//         for(int i=0;i<s.length();i++){
//             char ch = s.charAt(i);
//             map.put(ch,map.getOrDefault(ch,0)+1);
//         }
//         for(int i=0;i<t.length();i++){
//             char ch = t.charAt(i);
//             if(map.get(ch)!=null){
//                 if(map.get(ch) == 1){
//                     map.remove(ch);
//                 }else{
//                     map.put(ch,map.get(ch)-1);
//                 }
//             }else{
//                 return false;
//             }
//         }
//         return map.isEmpty();
//     }
// }

class Solution {
    public boolean isAnagram(String s, String t) {
      char[] arrs = s.toCharArray();
      char[] arrt = t.toCharArray();
      Arrays.sort(arrs);
      Arrays.sort(arrt);
      if(Arrays.equals(arrs,arrt)){
        return true;
      }else{
        return false;
      }
    }
}

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        seen = {}
        for ch in s:
            seen[ch] = seen.get(ch, 0) + 1
        for ch in t:
            seen[ch] = seen.get(ch, 0) - 1
            if seen[ch] < 0:
                return False
        return True
